Decorator/GuzzleClientFactoryDecorator.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Auxmoney\OpentracingGuzzleBundle\Decorator;
+
+use Auxmoney\OpentracingGuzzleBundle\Middleware\GuzzleRequestSpanning;
+use Auxmoney\OpentracingGuzzleBundle\Middleware\GuzzleTracingHeaderInjection;
+use GuzzleHttp\Client;
+use GuzzleHttp\HandlerStack;
+use Symfony\Component\DependencyInjection\Reference;
+
+final class GuzzleClientFactoryDecorator
+{
+
+    public static function decorate(
+        Client $client,
+        GuzzleRequestSpanning $spanning,
+        GuzzleTracingHeaderInjection $headerInjection
+    )
+    {
+        $handler = $client->getConfig("handler");
+        if ($handler instanceof HandlerStack) {
+            $handler->push($spanning);
```

Comment:
Should check for deduplication here

Reply:
Is this comment still valid?

Reply:
Nope, will ask @VincentLemaireAgriconomie to mark it as resolved